test/jasmine/tests/sliders_test.js

```diff
@@ -0,0 +1,234 @@
+                label: 'Label #1',
+                value: 'label-1'
+            }, {
+                method: 'update', args: [],
+                label: 'Label #2'
+            }, {
+                method: 'animate', args: [],
+                value: 'lacks-label'
+            }]
+        }];
+
+        supply(layoutIn, layoutOut);
+
+        expect(layoutOut.sliders[0].steps.length).toEqual(3);
+        expect(layoutOut.sliders[0].steps).toEqual([{
+            method: 'relayout', args: [],
+            label: 'Label #1',
+            value: 'label-1'
+        }, {
+            method: 'update', args: [],
+            label: 'Label #2',
+            value: 'Label #2'
+        }, {
+            method: 'animate', args: [],
+            label: 'step-2',
+            value: 'lacks-label'
+        }]);
+    });
+
+    it('should skip over non-object steps', function() {
+        layoutIn.sliders = [{
+            steps: [
+                null,
+                {
+                    method: 'relayout',
+                    args: ['title', 'Hello World']
+                },
+                'remove'
+            ]
+        }];
+
+        supply(layoutIn, layoutOut);
+
+        expect(layoutOut.sliders[0].steps.length).toEqual(1);
+        expect(layoutOut.sliders[0].steps[0]).toEqual({
+            method: 'relayout',
+            args: ['title', 'Hello World'],
+            label: 'step-1',
+            value: 'step-1',
+        });
+    });
+
+    it('should skip over steps with non-array \'args\' field', function() {
+        layoutIn.sliders = [{
+            steps: [{
+                method: 'restyle',
+            }, {
+                method: 'relayout',
+                args: ['title', 'Hello World']
+            }, {
+                method: 'relayout',
+                args: null
+            }, {}]
+        }];
+
+        supply(layoutIn, layoutOut);
+
+        expect(layoutOut.sliders[0].steps.length).toEqual(1);
+        expect(layoutOut.sliders[0].steps[0]).toEqual({
+            method: 'relayout',
+            args: ['title', 'Hello World'],
+            label: 'step-1',
+            value: 'step-1',
+        });
+    });
+
+    it('should keep ref to input update menu container', function() {
+        layoutIn.sliders = [{
+            steps: [{
+                method: 'relayout',
+                args: ['title', 'Hello World']
+            }]
+        }, {
+            bgcolor: 'red'
+        }, {
+            visible: false,
+            steps: [{
+                method: 'relayout',
+                args: ['title', 'Hello World']
+            }]
+        }];
+
+        supply(layoutIn, layoutOut);
+
+        expect(layoutOut.sliders[0]._input).toBe(layoutIn.sliders[0]);
+        expect(layoutOut.sliders[1]._input).toBe(layoutIn.sliders[1]);
+        expect(layoutOut.sliders[2]._input).toBe(layoutIn.sliders[2]);
+    });
+});
+
+describe('update sliders interactions', function() {
```

Comment:
Have you thought about adding mouse interaction tests?

It should be doable to mock a drag interaction using `new MouseEvent` - unless there are complications that I can't think about.